src/adapters/bridge-api.ts

```diff
@@ -0,0 +1,74 @@
+import axios from "axios";
+import { z } from "zod";
+
+import { StrictSchema } from "src/utils/type-safety";
+import * as domain from "src/domain";
+
+interface Bridge {
+  token_addr: string;
+  amount: string;
+  dest_net: number;
+  dest_addr: string;
+  deposit_cnt: string;
+}
+
+interface GetBridgesParams {
```

Comment:
I'd like to open a discussion regarding these whatever[Params] named interfaces.

Being interfaces to type the params object of some helper and accessory functions, would you agree on declaring them inline @elias-garcia and @adrifdez?

Reply:
What's the advantage of declaring them inline?

Reply:
> - Assure they are contextualized (i.e close to where they are used/required)

I don't see this a pro... At the end, if we do this always we would end up with `React.FC<{ // props }>` inline as well and much more things. The problem is the readability, take a look how the inline looks with more params and one more function:

<img width="238" alt="Captura de Pantalla 2022-04-04 a las 20 16 13" src="https://user-images.githubusercontent.com/6737805/161606774-502648e5-bcce-4cc0-8c41-89c60f098dde.png">

In comparison with declaring the interface on top:

<img width="466" alt="Captura de Pantalla 2022-04-04 a las 20 15 12" src="https://user-images.githubusercontent.com/6737805/161606855-ea4c7b1d-75c8-4472-9734-497506ed10ac.png">

Now imagine if you need to add more complex types like generics or using `&`, `|` operators...

> - Get better hints in VSCode

Using separate interfaces you'll have the same hints if you hit `ctrl + space`:

<img width="718" alt="Captura de Pantalla 2022-04-04 a las 20 13 11" src="https://user-images.githubusercontent.com/6737805/161607360-83f0fd83-cf57-4a80-8515-49ed45c6a905.png">

From my point of view I prefer to declare types and interfaces separately because they make the code more readable and easier to compose different types.

Reply:
This reads as if you were not using an object at all:

<img width="238" src="https://user-images.githubusercontent.com/6737805/161606774-502648e5-bcce-4cc0-8c41-89c60f098dde.png">

You see the name of the function and then, one param per line. The only difference is the `params` key which we could agree to call just `p` or `_`. It's by far more readable than this:

<img width="466" src="https://user-images.githubusercontent.com/6737805/161606855-ea4c7b1d-75c8-4472-9734-497506ed10ac.png">

Like this, I have to go to the interface `GetBridgesParams` to find the actual params wrapped in the object. Yes, I can use control and mitigate the problem for small, not deeply nested interfaces, but why having the problem at all?

The drawback of passing one object to the functions is precisely that we lose context and readability since we can no longer read the params in the context we need them: the signature of the function.